Traduction de la page "Uncontrolled Components" (#49)
reactjs/fr.reactjs.org: content/docs/uncontrolled-components.md

  }

  handleSubmit(event) {
    alert('Un nom a été soumis : ' + this.input.current.value);
    event.preventDefault();
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <label>
          Nom:
          <input type="text" ref={this.input} />
        </label>
        <input type="submit" value="Envoyer" />
      </form>
    );
  }
}
```

[**Essayer sur CodePen**](https://codepen.io/gaearon/pen/WooRWa?editors=0010)

> Puisqu'un composant non-contrôlé garde la source de vérité dans le DOM, il est parfois plus simple d'y intégrer du code React et non-React. Le code peut être également légèrement plus concis si vous souhaitez être rapide et grossier. Autrement, vous devriez utiliser les composants contrôlés.

Reviewer: je ne sais pas si grossier convient ici. plutot la traduction litterale, donc `sale` je pense

> Si quel type de composant utilisé pour une situation particulière n'est toujours pas clair, [cet article sur les entrées contrôlés ou non-contrôlés](http://goshakkk.name/controlled-vs-uncontrolled-inputs-react/) peut vous être utile.

Reviewer: pourquoi ne pas inverser la phrase `s'il n'est toujours pas clair quel type de composant doit etre utilisé dans une situation particulière, ...

Author: En effet, à la relecture, c'est plus léger d'inverser la phrase !

> Si quel type de composant utilisé pour une situation particulière n'est toujours pas clair, [cet article sur les entrées contrôlés ou non-contrôlés](http://goshakkk.name/controlled-vs-uncontrolled-inputs-react/) peut vous être utile.

Reviewer: cet article sur les entrées contrôlés ou non-contrôlés -> cet article sur les entrées contrôlées ou non-contrôlées
accorder avec entrées

> S'il n'est toujours pas clair quel type de composant doit être utilisé pour une situation particulière, [cet article sur les entrées contrôlées ou non-contrôlées](https://goshakkk.name/controlled-vs-uncontrolled-inputs-react/) peut vous être utile.

Reviewer 2: ```suggestion
Si dans une situation donnée il n'est pas toujours évident de savoir quel type de composant doit être utilisé, [cet article sur les entrées contrôlées ou non-contrôlées](https://goshakkk.name/controlled-vs-uncontrolled-inputs-react/) peut vous être utile.
```

Reviewer 2: Un petit passage à la voie active pour plus de fluidité ;)

> }

Author: Je ne suis pas sûr de comprendre ton commentaire ici. Le libellé de l'envoyeur c'est son nom non ?

> }

Reviewer 3: Je parle de la `value=` de l'`input type="submit"` ligne 61

```

De la même manière, `<input type="checkbox">` et `<input type="radio">` supportent l'attribut `defaultChecked`, tout comme `<select>` et `<textarea>` supportent `defaultValue`.

## La balise `input type="file"` {#the-file-input-tag}

> En HTML, un `<input type="file">` permet à l’utilisateur de sélectionner un ou plusieurs fichiers depuis son appareil, et de les téléverser vers un serveur ou de les manipuler en JavaScript grâce à [l'API File](https://developer.mozilla.org/en-US/docs/Web/API/File/Using_files_from_web_applications).

Reviewer 3: ```suggestion
En HTML, un `<input type="file">` permet à l’utilisateur de sélectionner un ou plusieurs fichiers depuis son appareil, et de les téléverser vers un serveur ou de les manipuler en JavaScript grâce à [l'API *File*](https://developer.mozilla.org/fr/docs/Web/API/File/Using_files_from_web_applications).
```

```html
<input type="file" />
```

> `<input type="file" />` est toujours un composant non-contrôlé en React, puisque sa valeur ne peut être définie que par un utilisateur et non programmatiquement.

Reviewer 3: ```suggestion
`<input type="file" />` est toujours un composant non-contrôlé en React, puisque sa valeur ne peut être définie que par l’utilisateur et non programmatiquement.
```

> Vous devriez utiliser l'API File pour interagir avec les fichiers. L'exemple suivant montre comment créer une [ref sur le noeud DOM](/docs/refs-and-the-dom.html) pour accéder aux fichiers dans un gestionnaire d'envoi.

Reviewer 3: ```suggestion
Vous devriez utiliser l'API *File* pour interagir avec les fichiers. L'exemple suivant montre comment créer une [ref sur le nœud DOM](/docs/refs-and-the-dom.html) pour accéder aux fichiers lors de l'envoi du formulaire :
```
